components/cqrs/command_processor.go

```diff
@@ -10,45 +10,117 @@ import (
 	"github.com/ThreeDotsLabs/watermill/message"
 )
 
-// CommandHandler receives a command defined by NewCommand and handles it with the Handle method.
-// If using DDD, CommandHandler may modify and persist the aggregate.
-//
-// In contrast to EventHandler, every Command must have only one CommandHandler.
-//
-// One instance of CommandHandler is used during handling messages.
-// When multiple commands are delivered at the same time, Handle method can be executed multiple times at the same time.
-// Because of that, Handle method needs to be thread safe!
-type CommandHandler interface {
-	// HandlerName is the name used in message.Router while creating handler.
+type CommandProcessorConfig struct {
+	// GenerateSubscribeTopic is used to generate topic for subscribing command.
+	GenerateSubscribeTopic CommandProcessorGenerateSubscribeTopicFn
+
+	// SubscriberConstructor is used to create subscriber for CommandHandler.
+	SubscriberConstructor CommandProcessorSubscriberConstructorFn
+
+	// OnHandle is called before handling command.
+	// OnHandle works in a similar way to middlewares: you can inject additional logic before and after handling a command.
+	//
+	// Because of that, you need to explicitly call params.Handler.Handle() to handle the command.
+	//   func(params CommandProcessorOnHandleParams) (err error) {
+	//       // logic before handle
+	//		 //  (...)
 	//
-	// It will be also passed to CommandsSubscriberConstructor.
-	// May be useful, for example, to create a consumer group per each handler.
+	//	     err := params.Handler.Handle(params.Message.Context(), params.Command)
```

Comment:
Something seems off with the formatting of the example overall (whitespace).

Reply:
can you explain more? 🤔 code blocks has extra indent to render godoc properly